template/requirements/make_base.py

```diff
@@ -43,7 +43,7 @@ def write_dependencies(dependency_name: str, dependencies: List[str]) -> None:
 with open("../pyproject.toml", "rb") as toml_file:
     pyproject = tomli.load(toml_file)
     dependencies = pyproject["project"].get("dependencies")
-    if not dependencies:
+    if dependencies is None:
```

Comment:
Why would you treat an empty list differently from not having a dependencies key at all? I'd just not raise in either case.

Reply:
I think I added this initially in case parsing "failed", indicating the pyproject is somehow written differently (don't know if it can).